src/main/java/uk/sky/cirrus/locking/CassandraLockingMechanism.java

```diff
@@ -0,0 +1,126 @@
+package uk.sky.cirrus.locking;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.exceptions.DriverException;
+import com.datastax.driver.core.exceptions.WriteTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.sky.cirrus.locking.exception.CannotAcquireLockException;
+import uk.sky.cirrus.locking.exception.CannotReleaseLockException;
+
+public class CassandraLockingMechanism extends LockingMechanism {
+
+    private static final Logger log = LoggerFactory.getLogger(CassandraLockingMechanism.class);
+
+    private final Session session;
+
+    private PreparedStatement insertLockQuery;
+    private PreparedStatement deleteLockQuery;
+    private boolean isRetryAfterWriteTimeout = false;
```

Comment:
Yep, the method using it used to be called in a loop, but now the loop is in this method we can do that.